react/components/SKUSelector/components/SelectVariationMode.tsx

```diff
@@ -24,6 +24,13 @@ function SelectVariationMode(props: VariationSelectModeProps) {
   const options = displayOptions.map(op => ({
     label: op.label,
     value: op.label,
+    disabled: op.disabled
+      ? op.impossible 
+        ? true
+        : !op.impossible && !op.available 
+        ? true 
+        : false
+      : false 
```

Comment:
Please break this into separate checks. It will greatly improve readability

Reply:
Added a separate function for readability - you can see it more clearly now that if it's disabled && impossible then it's true, if it's not impossible and not available as well then true, for any other case it's false.